src/app/qgisapp.cpp

```diff
@@ -6277,6 +6277,7 @@ void QgisApp::fileSaveAs()
     // add this to the list of recently used project files
     saveRecentProjectPath();
     mProjectLastModified = fullPath.lastModified();
+    emit newProject();
```

Comment:
I am a bit concerned about the use of newProject signal here - I would not expect it to be emitted and some parts of the app may use that signal to reset themselves to some default state. How about that the browser model listens to project singleton's signal when project writing is done to fix the underlying problem?

Reply:
I'm with @wonder-sk here too - this looks like abusing the existing signal.

Reply:
@nyalldawson @wonder-sk ok for not abusing the signal, but are you suggesting to move away from the existing pattern used here? 
https://github.com/qgis/QGIS/blob/master/src/app/qgisapp.cpp#L1065
Why not to add a new signal like `projectSaved(const QString &path )` to be used in this situation instead?
If not, then I think it would be better for consistency to also handle `newProject` connection to `QgsBrowserDockWidget::updateProjectHome` in the same way (in the model as you suggested).

Reply:
projectSaved signal does indeed sound much cleaner to me...!